### PR TITLE
Minor follow up fix for D/Objective-C

### PR DESCRIPTION
**Pull request by jacob-carlborg**

### CODE REVIEW

File: src/objc_stubs.c [hunk 1]
```diff
@@ -10,11 +10,12 @@
...
 #include "arraytypes.h"
-#include "class.c"
+#include "aggregate.h"
 #include "mars.h"
 #include "objc.h"
 #include "outbuffer.h"
 #include "parse.h"
+#include "scope.h"
 
 class ClassDeclaration;
```
Comment (rainers) [inline src/objc_stubs.c]: You can now remove some of these forward declarations.

File: src/objc_stubs.c [hunk 2]
```diff
@@ -10,11 +10,12 @@
  */
 
 #include "arraytypes.h"
-#include "class.c"
+#include "aggregate.h"
 #include "mars.h"
 #include "objc.h"
```
Comment (rainers) [inline src/objc_stubs.c]: objc.h currently does not compile as is, it depends on other includes before being included. I think you should add some more of the forward declarations in objc.h.

Comment (jacob-carlborg) [inline src/objc_stubs.c]: Like StructDeclaration? Anything else?

Comment (yebblies) [inline src/objc_stubs.c]: You should move the `#include "objc.h"` up to the top of the include list in `objc.c`, to prevent this from happening.

Comment (jacob-carlborg) [inline src/objc_stubs.c]: This is why I don't like C/C++. Can we just make the D port default and be done with it :smiley:

**Comment by ibuclaw** [comment]
There also looks to be some leaky abstractions, I'll let you know when I test with gdc.

**Comment by rainers** [comment]
I guess the build failures are caused by mixing struct/class forward declarations. Scope is a struct, not a class.
Otherwise LGTM.

**Comment by jacob-carlborg** [comment]
Thanks @rainers, that would probably have taken me some time to figure out.

**Comment by rainers** [comment]
Auto-merge toggled on

**Pull request merged**